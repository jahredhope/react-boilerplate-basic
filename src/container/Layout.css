

body {
  font-size: 18px;
  font-family: "Gill Sans Extrabold", Helvetica, sans-serif;
  background-color: var(--color-background);
}

input, select {
  font-size: inherit;
}

select {
  -webkit-appearance: none;
  -moz-appearance:    none;
  appearance:         none;
}

body {
  max-width: 100%;
  width: 100%;
  padding: 0;
  margin: 0;
  box-sizing: border-box;
}

:root {
  --bottom-margin: 20px;
  --page-max-width: 650px;

  --off-white: #fbfbfd;
  --dark-blue: #444488;
  --blue: #e0f0ff;
  --grey: #777777;
  --color-main: var(--blue);
  --color-trim: var(--dark-blue);
  --color-counter: var(--grey);
  --color-background: var(--off-white);
  --color-panel-background: white;

}


.container {
}
