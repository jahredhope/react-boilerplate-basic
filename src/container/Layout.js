import React, { Component, PropTypes} from 'react';
import { connect } from 'react-redux';
import Header from 'component/Header';
import NavMenu from 'component/NavMenu'
import Footer from 'component/Footer';

class Layout extends Component {
  render() {
    const {routes} = this.props;
    return (
      <div>
        <Header routes={routes} />
        <NavMenu />
        {this.props.children}
        <Footer />
      </div>
    );
  }
}


Layout.propTypes = {
  children: PropTypes.object.isRequired,
  routes: PropTypes.array.isRequired
};

export default connect()(Layout);
