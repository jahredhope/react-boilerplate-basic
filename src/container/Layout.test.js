import React from 'react';
import { Layout } from './Layout';
import renderer from 'react-test-renderer';

test('Layout sets a base Layout class and passes through children', () => {
  const component = renderer.create(
    <Layout />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
