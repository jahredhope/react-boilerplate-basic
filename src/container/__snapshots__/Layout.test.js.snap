exports[`test Layout sets a base Layout class and passes through children 1`] = `
<div
  className="mock-style-container">
  <header
    className="mock-style-container">
    <img
      className="mock-style-icon"
      height="50"
      src="test-file-stub" />
  </header>
  <nav
    className="mock-style-container">
    <a
      className="mock-style-item"
      href="/home"
      onClick={[Function]}
      style={Object {}}>
      Home
    </a>
    <a
      className="mock-style-item"
      href="/about"
      onClick={[Function]}
      style={Object {}}>
      About
    </a>
    <a
      className="mock-style-item"
      href="/thing"
      onClick={[Function]}
      style={Object {}}>
      Thing
    </a>
  </nav>
  <section>
    No route for that: Either we stuffed up a link or your being a shit
  </section>
  <footer
    className="mock-style-container">
    By Jahred Hope
  </footer>
</div>
`;
